tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -302,4 +302,144 @@ def create_or_delete_until_producer_fin():
+                               group=self.groups[idx],
+                               num_msgs=self.msg_count)
+
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        try:
+            producer.start(clean=False)
+            await_minimum_produced_records(self.redpanda,
+                                           producer,
+                                           min_acked=self.msg_count / 10)
+            await_s3_objects(self.redpanda)
+
+            consumer.start()
+            # Wait for consumer to fetch all data
+            consumer.wait()
+        except Exception as ex:
+            self._exceptions[idx] = ex
+            return
+
+        # The si-verifier (underneath FranzGoVerifier) writes non-printable bytes
+        # as the key and value. Check those.
+        key_size = 25
+
+        def check_msgs():
+
+            if len(consumer.messages) < self.msg_count:
+                return False
+
+            self.logger.debug(f'Len: {len(consumer.messages)}')
+
+            for msg in consumer.messages:
+                topic = msg['topic']
+                key = msg['key']
+                value = msg['value']
+
+                if topic != self.topics[idx].name:
+                    self.logger.debug(f'Topic check failed: {topic}')
+                    return False
+
+                if len(key) != key_size:
+                    self.logger.debug(
+                        f'Key check failed: Expected size {key_size}, got size {len(key)}'
+                    )
+                    return False
+
+                if len(value) != self.msg_size:
+                    self.logger.debug(
+                        f'Value check failed: Expected size {self.msg_size}, got size {len(value)}'
+                    )
+                    return False
+
+            return True
+
+        try:
+            wait_until(check_msgs,
+                       timeout_sec=300,
+                       backoff_sec=1,
+                       err_msg='consumer failed to fetch all messages')
+        except Exception as ex:
+            self._exceptions[idx] = ex
+
+        producer.stop()
+        consumer.stop()
+
+    @cluster(num_nodes=7, log_allow_list=KGO_ALLOW_LOGS)
+    def test_consumer_groups(self):
+
+        await_bucket_creation(self.redpanda,
+                              self._si_settings.cloud_storage_bucket)
+
+        # Remote write/read and retention set at topic level
+        rpk = RpkTool(self.redpanda)
+        for spec in self.topics:
+            rpk.alter_topic_config(spec.name, 'redpanda.remote.write', 'true')
+            rpk.alter_topic_config(spec.name, 'redpanda.remote.read', 'true')
+            rpk.alter_topic_config(spec.name, 'retention.bytes',
+                                   str(self.segment_size))
+
+        workers = []
+        for idx in range(self.num_groups):
```

Comment:
From a quick read, I think this is creating a bunch of topics with one consumer group each, and each consumer group has one node in it.

For a test that's specifically exercising consumer groups, I think it would make sense to have multiple consumers per group, even if that means just using one topic.